keyboard.js

```diff
@@ -0,0 +1,167 @@
+const Keyboard = {
+    elements: {
+        main: null,
+        keysContainer: null,
+        keys: []
+    },
+
+    eventHandlers: {
+        oninput: null,
+        onclose: null
+    },
+
+    properties: {
+        value: "",
+        capsLock: false
+    },
+
+    init() {
+
+        //create main elements
+        this.elements.main = document.createElement('div');
+        this.elements.keysContainer = document.createElement('div');
+
+        //setup main elements
+        this.elements.main.classList.add('keyboard', 'keyboard--hidden');
+        this.elements.keysContainer.classList.add('keyboard__keys');
+        this.elements.keysContainer.appendChild(this._createKeys());
+
+        this.elements.keys = this.elements.keysContainer.querySelectorAll('.keyboard__key');
+
+        //add to DOM
+        this.elements.main.appendChild(this.elements.keysContainer);
+        document.body.appendChild(this.elements.main);
+
+        // Auto use keyboard
+        document.querySelectorAll('.use-keyboard-input').forEach(element => {
+            element.addEventListener('focus', () => {
+                this.open(element.value, currentValue => {
+                    element.value = currentValue;
+                });
+            })
+        })
+
+        const parentForKeyboard = document.querySelector('.keyboard__keys');
+        parentForKeyboard.addEventListener('click', event => {
+            if (Object.values(event.target.parentNode.classList).indexOf('keyboard__key') === -1 && Object.values(event.target.classList).indexOf('keyboard__key') === -1) {
```

Comment:
1) you performs these 2 actions too many time. Put them in variables
2) why don't you use element.classList.contains(class)? Why do you use such complex construction Object.values(event.target.parentNode.classList).indexOf('keyboard__key')?